server.go

```diff
@@ -1644,3 +1642,22 @@ func (s *server) Peers() []*peer {
 
 	return peers
 }
+
+// SetAlias updates the server's alias and broadcasts a new node announcement
+//
+// NOTE: This function is safe for concurrent access.
+func (s *server) SetAlias(alias string) error {
+	nodeAlias, err := lnwire.NewNodeAlias(alias)
+	if err != nil {
+		return err
+	}
+	s.currentNodeAnn.Alias = nodeAlias
```

Comment:
this needs to acquire the server mutex before modifying the node announcement to actually be thread safe.

Reply:
actually, it won't work setting the alias directly like this, because this would invalidate the signature. You will need to resign the `currentNodeAnn`.

Reply:
Whoops, forgot to claim the mutex. 

For your second point, it looks like `genNodeAnnouncement(true)` resigns the node announcement. I can explicitly resign the announcement after updating the alias if that's preferred.

Reply:
Two problems with this:
1) `genNodeAnnouncement` will also claim the mutex, so this will deadlock.
2) if you set the alias, release the mutex, then call `genNodeAnnouncement`, some other go routine might get the node announcement in the meantime, but it will be invalid because of the changed alias. So you must make sure the alias change and the resigning happens under the same mutex.